### PR TITLE
Deprecate the `tekton-provenance` predicate format

### PR DESCRIPTION
**Pull request by priyawadhwa**
Added in a doc to describe how we plan to deprecate features. We can make an announcement for deprecation and then actually remove the feature two releases later.

For now, this will still work but we just print the error message.
It can be removed in a future release, and I removed all docs around it as well.
https://github.com/tektoncd/chains/issues/293
ref https://github.com/tektoncd/chains/issues/249

### CODE REVIEW

**Comment by priyawadhwa** [comment]
Hey @lukehinds thanks for the lgtm! I think prow also needs an `/approved` for this to merge 😅

File: docs/deprecations.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- [Introduction](#introduction)
+- [Deprecation Table](#deprecation-table)
+
+## Introduction
+
+This doc provides a list of features in Tekton Chains that are
+being deprecated.
+
+Deprecations will follow this timeline:
+- Deprecation Announcement is made during a release
```
Comment (loosebazooka) [inline docs/deprecations.md:19]: lower case `a`nnouncement

File: docs/deprecations.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+So, if a feature is deprecated at v0.1.0, then it would be removed in v0.3.0.
+
+## Deprecation Table
+
+| Feature Being Deprecated                                                                                                                                            | Deprecation Announcement                                             | API Compatibility Policy | Earliest Date or Release of Removal |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | ----------------------------------- |
+| [`tekton-provenance` format is deprecated](https://github.com/tektoncd/chains/issues/293)                                                                         | [v0.6.0](https://github.com/tektoncd/pipeline/releases/tag/v0.6.0) | Alpha                                                                                                   | v0.8.0                   |
```
Comment (loosebazooka) [inline docs/deprecations.md:28]: make this table a little bit easier to maintain?
```suggestion
| Feature | Deprecation Announcement | API Compatibility Policy | Earliest Date or Release of Removal |
| --- | --- | --- | --- |
| [`tekton-provenance` format ](https://github.com/tektoncd/chains/issues/293) | [v0.6.0](https://github.com/tektoncd/pipeline/releases/tag/v0.6.0) | Alpha | v0.8.0 |
```

Comment (priyawadhwa) [reply docs/deprecations.md]: good call, fixed!

**Pull request merged**